lib/netsuite/async_result.rb

```diff
@@ -0,0 +1,59 @@
+# https://system.netsuite.com/help/helpcenter/en_US/Output/Help/SuiteCloudCustomizationScriptingWebServices/SuiteTalkWebServices/AsynchronousRequestProcessing.html
+module NetSuite
+  class AsyncResult
+
+    attr_reader :errors, :type
+
+    def initialize(async_result)
+      @type = async_result[:"@xsi:type"]
+
+      base_ref_list = []
+      errors = []
+
+      if async_result[:read_response_list]
+        response_list = async_result[:read_response_list][:read_response]
+      else
+        response_list = async_result[:write_response_list][:write_response]
+      end
+
+      Array[response_list].flatten.each do |response|
+        if response[:status][:@is_success] == 'true'
```

Comment:
right. What's the point of using !! then? I thought it was used in this case. Wow, fail on my part. Good catch @jeromecornet